test/Microsoft.Diagnostics.EventFlow.Outputs.Tests/EventHubOutputTests.cs

```diff
@@ -55,26 +55,68 @@ public void UsesIsoDateFormat()
             e.Payload.Add("StringProperty", "perfection");
 
             EventHubOutput eho = new EventHubOutput(configuration, healthReporter.Object, connectionString => client.Object);
-            await eho.SendEventsAsync(new EventData[] { e }, 17, CancellationToken.None);
+            await eho.SendEventsAsync(new EventData[] {e,}, 17, CancellationToken.None);
 
-            Func<IEnumerable<MessagingEventData>, bool> verifyBatch = batch => {
+            Func<IEnumerable<MessagingEventData>, bool> verifyBatch = batch =>
+            {
                 if (batch.Count() != 1) return false;
 
                 var data = batch.First();
                 var bodyString = Encoding.UTF8.GetString(data.Body.Array, data.Body.Offset, data.Body.Count);
                 var recordSet = JObject.Parse(bodyString);
                 var message = recordSet["records"][0];
+
                 return (string) message["level"] == "Warning"
-                    && (string) message["properties"]["ProviderName"] == "TestProvider"
-                    && (int) message["properties"]["IntProperty"] == 42
-                    && (string) message["properties"]["StringProperty"] == "perfection";
+                       && (string) message["properties"]["ProviderName"] == "TestProvider"
+                       && (int) message["properties"]["IntProperty"] == 42
+                       && (string) message["properties"]["StringProperty"] == "perfection";
             };
 
             client.Verify(c => c.SendAsync(It.Is<IEnumerable<MessagingEventData>>(b => verifyBatch(b))), Times.Once);
             healthReporter.Verify(hr => hr.ReportWarning(It.IsAny<string>(), It.IsAny<string>()), Times.Never);
             healthReporter.Verify(hr => hr.ReportProblem(It.IsAny<string>(), It.IsAny<string>()), Times.Never);
         }
-    }
 
-    
-}
+        [Fact]
+        public async Task SendsDataToPartitionedEventHub()
```

Comment:
Cool. Can you please add one more test case that tests the following scenario: 

1. 3 EventData objects
2. One has PartitionByProperty == "A"
3. Second has PartitionByProperty == "B"
4. Third does not have a PartitionByProperty

.. and verify that expected IEventHubClient overloads are called with partition key set to the right values (we expect 3 different calls here).

After that is done it is good to go!

Reply:
@karolz-ms I extended the tests to test these three cases (2 partitionKey values plus event without PartitionByProperty value). Also I added three events with big payload targeting same "Partiton2" value for PartitionByProperty, so also the batch splitting refactoring I applied is tested.